scipy/io/matlab/tests/test_streams.py

```diff
@@ -27,82 +28,70 @@
     _read_into, _read_string
 
 
-fs = None
-gs = None
-cs = None
-fname = None
-
-
-def setup():
+@contextmanager
+def setup_test_file():
     val = b'a\x00string'
-    global fs, gs, cs, fname
     fd, fname = mkstemp()
-    fs = os.fdopen(fd, 'wb')
-    fs.write(val)
-    fs.close()
-    fs = open(fname, 'rb')
-    gs = BytesIO(val)
-    cs = cStringIO(val)
-
-
-def teardown():
-    global fname, fs
-    fs.close()
-    del fs
+    with os.fdopen(fd, 'wb') as fs:
+        fs.write(val)
+    with open(fname, 'rb') as fs:
+        gs = BytesIO(val)
+        cs = cStringIO(val)
+        yield fs, gs, cs
     os.unlink(fname)
 
 
 def test_make_stream():
-    global fs, gs, cs
-    # test stream initialization
-    assert_(isinstance(make_stream(gs), GenericStream))
-    if sys.version_info[0] < 3:
-        assert_(isinstance(make_stream(cs), cStringStream))
-        assert_(isinstance(make_stream(fs), FileStream))
+    with setup_test_file() as (fs, gs, cs):
```

Comment:
These means the tempfile is created three times during testing, does not seem so bad to me.